pkg/controllers/controllers.go

```diff
@@ -35,14 +35,15 @@ import (
 	nodeclasstermination "github.com/Azure/karpenter-provider-azure/pkg/controllers/nodeclass/termination"
 
 	"github.com/Azure/karpenter-provider-azure/pkg/controllers/nodeclaim/inplaceupdate"
+	"github.com/Azure/karpenter-provider-azure/pkg/providers/imagefamily"
 	"github.com/Azure/karpenter-provider-azure/pkg/providers/instance"
 )
 
 func NewControllers(ctx context.Context, mgr manager.Manager, kubeClient client.Client, recorder events.Recorder,
-	cloudProvider cloudprovider.CloudProvider, instanceProvider instance.Provider) []controller.Controller {
+	cloudProvider cloudprovider.CloudProvider, instanceProvider instance.Provider, imageProvider *imagefamily.Provider) []controller.Controller {
```

Comment:
`imagefamily.Provider` should be an interface. Generally speaking accept interfaces, return structs (that's the Go way).

This function is currently follow that rule, IMO you should maintain that.

Reply:
Yea, this is mostly an artifct of waiting on a folllow up PR where I was planning on doing some of this refactoring, but this change can go in here. Just need to break this input param into the two seperate interfaces for k8s, and image.

Reply:
Updated